cirq-core/cirq/protocols/json_serialization.py

```diff
@@ -283,9 +283,14 @@ def _json_dict_with_cirq_type(obj: Any):
     if 'cirq_type' in base_dict:
         # TODO: upgrade to ValueError in v0.15
         warnings.warn(
-            f"Found 'cirq_type': '{base_dict['cirq_type']}' in _json_dict_. "
-            f"Custom values of this field are not permitted, and will produce "
-            "an error starting in Cirq v0.15.",
+            f"Found 'cirq_type': '{base_dict['cirq_type']}' in user-specified _json_dict_. "
+            "This field is now automatically generated from the class name and (if present) "
+            "its _json_namespace_ method. Starting in v0.15, custom 'cirq_type' values "
+            "will trigger an error."
+            "\n\n"
+            "To fix this, remove 'cirq_type' from the class _json_dict_ method. Classes "
```

Comment:
Should we also mention that serialized objects will continue to be the same as previously serialized objects?

Reply:
Emphasized this in the preceding paragraph. FWIW, this only preserves serialization format if a class follows the `[namespace.]classname` convention, but we've been _exceedingly_ consistent about applying that convention in Cirq. Hopefully that translates to external users doing the same.